pages/_app.tsx

```diff
@@ -180,14 +217,29 @@ const Analytics = () => {
 
 const MyApp = ({ Component, pageProps }: AppProps) => {
   const router = useRouter();
+  const isEngaged = useIsEngaged();
 
   useEffect(() => {
-    posthog(); // init posthog
+    if (!isEngaged) return;
+    // Trigger engagement view events here
+    sendEngagedView();
+  }, [isEngaged]);
```

Comment:
Do we have to return an `isEngaged`? Would it be better if we just `sendEngagedView` in the hook where we are setting engaged?

Reply:
I think this mainly comes down to which solution you perceive as being more readable. I think the current solution keeps the hook readable by isolating the tracking logic from the tracking events. 
But like I said, it's mostly personal preference at this point.